# Publications 

[Google Scholar](https://scholar.google.com/citations?user=A2ImcaUAAAAJ&hl=en)

* **Weaver**, S., Shank, S. D., Spielman, S. J., Li, M., Muse, S. V., & Kosakovsky Pond, S. L. (2018). Datamonkey 2.0: a modern web application for characterizing selective and other evolutionary processes. Molecular Biology and Evolution, 35(3), 773–777.
* Viana, R., Moyo, S., Amoako, D. G., Tegally, H., Scheepers, C., Althaus, C. L., Anyaneji, U. J., Bester, P. A., Boni, M. F., Chand, M., & others. (2022). Rapid epidemic expansion of the SARS-CoV-2 Omicron variant in southern Africa. Nature, 1–10.
* Martin, D. P., Lytras, S., Lucaci, A. G., Maier, W., Gruning, B., Shank, S. D., **Weaver**, S., MacLean, O. S., Orton, R. J., Lemey, P., & others. (2022). Selection analysis identifies unusual clustered mutational changes in Omicron lineage BA. 1 that likely impact Spike function. BioRxiv.
* Caraballo-Ortiz, M. A., Miura, S., Sanderford, M., Dolker, T., Tao, Q., **Weaver**, S., Pond, S. L. K., & Kumar, S. (2022). TopHap: rapid inference of key phylogenetic structures from common haplotypes in large genome collections with limited diversity. Bioinformatics, 38(10), 2719–2726. https://doi.org/10.1093/bioinformatics/btac186
* Kumar, S., Tao, Q., **Weaver**, S., Sanderford, M., Caraballo-Ortiz, M. A., Sharma, S., Pond, S. L., & Miura, S. (2021). An evolutionary portrait of the progenitor SARS-CoV-2 and its dominant offshoots in COVID-19 pandemic. Molecular Biology and Evolution, 38(8), 3046–3059.
* Tegally, H., Wilkinson, E., Giovanetti, M., Iranzadeh, A., Fonseca, V., Giandhari, J., Doolabh, D., Pillay, S., San, E. J., Msomi, N., & others. (2021). Detection of a SARS-CoV-2 variant of concern in South Africa. Nature, 592(7854), 438–443.
* Lucaci, A. G., Wisotsky, S. R., Shank, S. D., **Weaver**, S., & Kosakovsky Pond, S. L. (2021). Extra base hits: widespread empirical support for instantaneous multiple-nucleotide changes. PloS One, 16(3), e0248337.
* Lucaci, A. G., Zehr, J. D., Shank, S. D., **Weaver**, S., Bouvier, D., Mei, H., Nekrutenko, A., & Pond, S. L. K. (n.d.). Rapid molecular sequence characterization of the SARS-CoV-2 variant of interest Lambda (C. 37).
* MacLean, O. A., Lytras, S., **Weaver**, S., Singer, J. B., Boni, M. F., Lemey, P., Kosakovsky Pond, S. L., & Robertson, D. L. (2021). Natural selection in the evolution of SARS-CoV-2 in bats created a generalist virus and highly capable human pathogen. PLoS Biology, 19(3), e3001115.
* Maher, M. C., Bartha, I., **Weaver**, S., Di Iulio, J., Ferri, E., Soriaga, L., Lempp, F. A., Hie, B. L., Bryson, B., Berger, B., & others. (2021). Predicting the mutational drivers of future SARS-CoV-2 variants of concern. Science Translational Medicine, eabk3445.
* Martin, D. P., **Weaver**, S., Tegally, H., San, J. E., Shank, S. D., Wilkinson, E., Lucaci, A. G., Giandhari, J., Naidoo, S., Pillay, Y., & others. (2021). The emergence and ongoing convergent evolution of the SARS-CoV-2 N501Y lineages. Cell, 184(20), 5189–5200.
* Cele, S., Gazy, I., Jackson, L., Hwa, S.-H., Tegally, H., Lustig, G., Giandhari, J., Pillay, S., Wilkinson, E., Naidoo, Y., & others. (2021). Escape of SARS-CoV-2 501Y. V2 from neutralization by convalescent plasma. Nature, 593(7857), 142–146.
* Fischer, W., Giorgi, E. E., Chakraborty, S., Nguyen, K., Bhattacharya, T., Theiler, J., Goloboff, P. A., Yoon, H., Abfalterer, W., Foley, B. T., & others. (2021). HIV-1 and SARS-CoV-2: Patterns in the evolution of two pandemic pathogens. Cell Host & Microbe, 29(7), 1093–1110.
* Kosakovsky Pond, S. L., Wisotsky, S. R., Escalante, A., Magalis, B. R., & **Weaver**, S. (2021). Contrast-FEL—a test for differences in selective pressures at individual sites among clades and sets of branches. Molecular Biology and Evolution, 38(3), 1184–1198.
* Dávila-Conn, V., García-Morales, C., Matías-Florentino, M., López-Ortiz, E., Paz-Juárez, H. E., Beristain-Barreda, Á., Cárdenas-Sandoval, M., Tapia-Trejo, D., López-Sánchez, D. M., Becerril-Rodríguez, M., & others. (2021). Characteristics and growth of the genetic HIV transmission network of Mexico City during 2020. Journal of the International AIDS Society, 24(11), e25836.
* Baker, D., Van Den Beek, M., Blankenberg, D., Bouvier, D., Chilton, J., Coraor, N., Coppens, F., Eguinoa, I., Gladman, S., Grüning, B., & others. (2020). No more business as usual: Agile and effective responses to emerging pathogen threats require open data and open analytics. PLoS Pathogens, 16(8), e1008643.
* Spielman, S. J., **Weaver**, S., Shank, S. D., Magalis, B. R., Li, M., & Kosakovsky Pond, S. L. (2019). Evolution of viral genomes: interplay between selection, recombination, and other forces. In Evolutionary Genomics (pp. 427–468). Humana, New York, NY.
* Shank, S. D., **Weaver**, S., & Kosakovsky Pond, S. L. (2018). phylotree. js-a JavaScript library for application development and interactive data visualization in phylogenetics. BMC Bioinformatics, 19(1), 1–5.
* Eren, K., **Weaver**, S., Ketteringham, R., Valentyn, M., Laird Smith, M., Kumar, V., Mohan, S., Kosakovsky Pond, S. L., & Murrell, B. (2018). Full-Length Envelope Analyzer (FLEA): A tool for longitudinal analysis of viral amplicons. PLoS Computational Biology, 14(12), e1006498.
* Kosakovsky Pond, S. L., **Weaver**, S., Leigh Brown, A. J., & Wertheim, J. O. (2018). HIV-TRACE (TRAnsmission Cluster Engine): a tool for large scale molecular epidemiology of HIV-1 and other rapidly evolving pathogens. Molecular Biology and Evolution, 35(7), 1812–1819.
* Laird Smith, M., Murrell, B., Eren, K., Ignacio, C., Landais, E., **Weaver**, S., Phung, P., Ludka, C., Hepler, L., Caballero, G., & others. (2016). Rapid sequencing of complete env genes from primary HIV-1 samples. Virus Evolution, 2(2).
* Smith, M. D., Wertheim, J. O., **Weaver**, S., Murrell, B., Scheffler, K., & Kosakovsky Pond, S. L. (2015). Less is more: an adaptive branch-site random effects model for efficient detection of episodic diversifying selection. Molecular Biology and Evolution, 32(5), 1342–1353.
* Murrell, B., **Weaver**, S., Smith, M. D., Wertheim, J. O., Murrell, S., Aylward, A., Eren, K., Pollner, T., Martin, D. P., Smith, D. M., & others. (2015). Gene-wide identification of episodic selection. Molecular Biology and Evolution, 32(5), 1365–1371.
* Hepler, N. L., Scheffler, K., **Weaver**, S., Murrell, B., Richman, D. D., Burton, D. R., Poignard, P., Smith, D. M., & Kosakovsky Pond, S. L. (2014). IDEPI: rapid prediction of HIV-1 antibody epitopes and other phenotypic features from sequence data using a flexible machine learning platform. PLoS Computational Biology, 10(9), e1003842.
* Young, J., Mote, D., Martinez, M., Rodriguez-Mueller, J., **Weaver**, S., Smith, D., Richman, D., Little, S., & Pond, S. K. (n.d.). OCCAMS: An Open Source Clinical Content Analysis and Management System. Applications for HIV Primary Infection Research.
